model/task/db.go

```diff
@@ -594,6 +595,13 @@ func GetRecentTasks(period time.Duration) ([]Task, error) {
 		return nil, errors.Wrap(err, "problem with stats query")
 	}
 
+	for i, task := range out {
+		if err := task.MergeNewTestResults(); err != nil {
+			return nil, errors.Wrap(err, "error merging new test results")
+		}
+		out[i] = task
+	}
```

Comment:
this will result in duplicate tests?

Reply:
I think this can only happen if we are in a state where a task is logged to the DB using both the old and new data structures right? Do we expect to be in that state at any point?

Reply:
What will?

Reply:
We don't expect to ever be in that state.

Reply:
I think it depends on how the migration works? 

Do we copy the data over in one migration and then unset the field in another migration? If so then it's duplicated. Because we can't do the operation in a transaction, during the migration process it'll also be duplicated. 

I think its safe to write the TestResult field in the Task struct with the "joined" content, ignoring any data that might have been in that field?

Reply:
Fixed.